src/pudl/transform/vcerare.py

```diff
@@ -283,107 +289,232 @@ def out_vcerare__hourly_available_capacity_factor(
-    blocking=True,
-    description="Check that output table is as expected.",
-    op_tags={"memory-use": "high"},
-)
-def check_hourly_available_cap_fac_table(asset_df: pd.DataFrame):  # noqa: C901
-    """Check that the final output table is as expected."""
+@asset
+def out_vcerare__hourly_available_capacity_factor(
+    raw_vcerare__lat_lon_fips: pd.DataFrame,
+    raw_vcerare__fixed_solar_pv_lat_upv: pd.DataFrame,
+    raw_vcerare__offshore_wind_power_140m: pd.DataFrame,
+    raw_vcerare__onshore_wind_power_100m: pd.DataFrame,
+):
+    """Transform raw Vibrant Clean Energy renewable generation profiles.
+
+    Concatenates the solar and wind capacity factors into a single table and turns
+    the columns for each county or subregion into a single county_or_lake_name column.
+    Graph asset will process 1 year of data at a time to limit peak memory usage.
+    """
+
+    def _get_year(df, year):
+        return df.loc[df["report_year"] == year]
+
+    for year in raw_vcerare__fixed_solar_pv_lat_upv["report_year"].unique():
+        one_year_hourly_available_capacity_factor(
+            year=year,
+            raw_vcerare__lat_lon_fips=raw_vcerare__lat_lon_fips,
+            raw_vcerare__fixed_solar_pv_lat_upv=_get_year(
+                raw_vcerare__fixed_solar_pv_lat_upv, year
+            ),
+            raw_vcerare__offshore_wind_power_140m=_get_year(
+                raw_vcerare__offshore_wind_power_140m, year
+            ),
+            raw_vcerare__onshore_wind_power_100m=_get_year(
+                raw_vcerare__onshore_wind_power_100m, year
+            ),
+        )
+
+
+def _check_rows(vce) -> AssetCheckResult | None:
     logger.info("Check VCE RARE hourly table is the expected length")
-    if (length := len(asset_df)) != 136437000:
+    (length,) = duckdb.query("SELECT COUNT(*) FROM vce").fetchone()
+    if length != 136437000:
         return AssetCheckResult(
             passed=False,
             description="Table unexpected length",
             metadata={"table_length": length, "expected_length": 136437000},
         )
+    return None
+
+
+def _check_nulls(vce) -> AssetCheckResult | None:
     logger.info("Check there are no NA values in VCE RARE table (except FIPS)")
-    if asset_df[asset_df.columns.difference(["county_id_fips"])].isna().any().any():
-        return AssetCheckResult(
-            passed=False,
-            description="Found NA values when there should be none",
-            metadata={
-                "NA index values": asset_df[asset_df.isna().any(axis="columns")].index
-            },
-        )
+    columns = [c for c in vce.columns if c != "county_id_fips"]
+    for c in columns:
+        nulls = duckdb.query(f"SELECT {c} FROM vce WHERE {c} IS NULL").fetchall()  # noqa: S608
+        if len(nulls) > 0:
+            return AssetCheckResult(
+                passed=False,
+                description=f"Found NA values in column {c}",
+            )
```

Comment:
Rather than bailing as soon as we get any unexpected nulls, can we check all the columns, and tell the user what columns had unexpected nulls? Will make debugging less frustrating.

Reply:
Fixed to find all null columns